yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -184,6 +184,9 @@ private[spark] class ApplicationMaster(
     try {
       val appAttemptId = client.getAttemptId()
 
+      var context = s"Spark_AppName_${System.getProperty("spark.app.name")}" +
```

Comment:
it would be nice to have a class or at least util function to construct the context string so that we don't have same string in multiple places.